src/main/scala/Engine.scala

```diff
@@ -36,12 +37,17 @@ case class Query(
     fields: Option[List[Field]] = None, // default: whatever is in algorithm params or None
     currentDate: Option[String] = None, // if used will override dateRange filter, currentDate must lie between the item's
     // expireDateName value and availableDateName value, all are ISO 8601 dates
-    dateRange: Option[DateRange] = None, // optional before and after filter applied to a date field
-    blacklistItems: Option[List[String]] = None, // default: whatever is in algorithm params or None
+    dateRange: Option[DateRange] = None, // default: whatever is in algorithm params or None
     returnSelf: Option[Boolean] = None,// means for an item query should the item itself be returned, defaults
                                        // to what is in the algorithm params or false
     num: Option[Int] = None, // default: whatever is in algorithm params, which itself has a default--probably 20
-    eventNames: Option[List[String]]) // names used to ID all user actions
+    eventNames: Option[List[String]], // names used to ID all user actions
+    /** following variable defines features - must include particular items, price range, blacklist items,blacklistCategories respectively ***/
+    includeItems: Option[List[String]] = None,
+    priceRange: Option[PriceRange] = None,// optional before and after filter applied to a date field
+    blacklistItems: Option[List[String]] = None,
+    blacklistCategory: Option[List[String]] = None,
+    blacklistBrand: Option[List[String]] = None)
   extends Serializable
```

Comment:
We need to be able to specify a filter like we do with properties, which include the property name so they can be used for any property not just Category or Brand. I suggest an addition to the field definition, perhaps a bias enumeration, like 

"bias": {
    "type": "not"
}

"bias": {
    "value": -1 // for filter
}

the type and value would not be used together.

by default the type is None, which means current behavior.

This will allow any number of fields to be set for any named property. Likewise this could be put in the engine.json for global application or into the query.